### DISCUSSION TITLE
Do we have a recommended base image for new containers?

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io#111] [Q&A] [answered]
Hi @stebo85 @aswinnarayanan 

I noticed that in the current template, the base image is neurodebian rather than bare ubuntu like we used previously:
https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/template/build.sh

Is that the recommendation for new containers? Do we have such a recommendation or it doesn't really matter?

I'm asking because many times people build new versions of old containers (e.g., I built bidscoin 3.7.0 instead of version 2.X that was available before), and if we do not instruct them, they might just keep the same base image as in the original container. This might not be what we want.

Cheers,
Oren

**Comment by stebo85** [comment, marked as answer]
neurodebian is a good starting point as it makes it very easy to install neuroimaging software, but the base image will need to be adjusted if it doesn't work for certain applications. Ultimately, it doesn't matter which base image is used as long as the application works.

**Comment by civier** [comment]
Ok. thanks for the info.

Regarding Singularity container nesting, I remember that it requires user namespace, and this is not supported by all version of Linux. So maybe we can at least have a recommendation of minimum versions of Linux to use -- it will then be easier to enable container nesting when we need to.
Do you have the minimum versions that support user namespace handy? (maybe for Ubuntu, CentOs, Neurodebian, which will be the most popular) If not, I can look it up myself.

**Reply by stebo85** [reply]
@civier,

We already got container nesting to work and we are using it in the code container: https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/code/build.sh

It is always advisable to use the latest operating system that's available and that supports the application for security reasons and this should be the recommendation.

**Comment by civier** [comment]
Great. Happy to hear that it's already working. Wasn't sure.

Container nesting is something that I will explore also for the MATLAB container, so would be great if you could document in https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/code/build.sh
what had to be included for nesting to work.

If you already explained everything in an issue, you're welcome to link there. I might overlooked it when I was busy.

**Reply by stebo85** [reply]
This is the issue: https://github.com/NeuroDesk/neurodesktop/issues/20 and the code example (https://github.com/NeuroDesk/neurocontainers/blob/master/recipes/code/build.sh) is currently the documentation until we tested it enough.
